packages/core/src/delegation.js

```diff
@@ -192,12 +194,20 @@ export class Delegation {
     Object.defineProperties(this, { data: { value: data, enumerable: false } })
     return data
   }
+  /**
+   * Attach a block to the invocation encoded bytes.
```

Comment:
```suggestion
   * Attach a block to the invocation DAG so it would be included in the
   * block iterator.
   * ⚠️ You should only attach blocks that are referenced from the `capabilities`
   * or `facts`, if that is not the case you probably should reconsider.
```

Nit: I think this is more clear